From 3ba9712890f82a02acd4f49edc1c6ec5429a179f Mon Sep 17 00:00:00 2001
From: Nucleus Build <nucleus@localhost>
Date: Fri, 24 Jul 2026 22:30:00 -0600
Subject: [PATCH] Export Composer3 framework seam for Nucleus

Keep the AIDL command engine shared while allowing the Nucleus product to own
its display discovery and frame-composition backend. Keep the reusable
interfaces independent of the ranchu DRM implementation.
---
 hals/hwc3/Android.bp        | 17 +++++++++++++++++
 hals/hwc3/ComposerClient.cpp |  2 --
 hals/hwc3/DisplayFinder.h   |  3 ++-
 hals/hwc3/FrameComposer.h   |  2 +-
 4 files changed, 20 insertions(+), 4 deletions(-)

diff --git a/hals/hwc3/Android.bp b/hals/hwc3/Android.bp
index b4494fc..98fa154 100644
--- a/hals/hwc3/Android.bp
+++ b/hals/hwc3/Android.bp
@@ -22,6 +22,23 @@ package {
     ],
 }
 
+filegroup {
+    name: "nucleus_hwc3_common_sources",
+    srcs: [
+        "Common.cpp",
+        "Composer.cpp",
+        "ComposerClient.cpp",
+        "ComposerResources.cpp",
+        "Display.cpp",
+        "DisplayConfig.cpp",
+        "Layer.cpp",
+        "VsyncThread.cpp",
+    ],
+    visibility: [
+        "//device/nucleus/nucleus_x86_64:__pkg__",
+    ],
+}
+
 cc_binary {
     name: "android.hardware.graphics.composer3-service.ranchu",
 
diff --git a/hals/hwc3/ComposerClient.cpp b/hals/hwc3/ComposerClient.cpp
index 0a68cb1..039eb8e 100644
--- a/hals/hwc3/ComposerClient.cpp
+++ b/hals/hwc3/ComposerClient.cpp
@@ -21,8 +21,6 @@
 
 #include "Common.h"
 #include "Device.h"
-#include "GuestFrameComposer.h"
-#include "HostFrameComposer.h"
 #include "Time.h"
 
 namespace aidl::android::hardware::graphics::composer3::impl {
diff --git a/hals/hwc3/DisplayFinder.h b/hals/hwc3/DisplayFinder.h
index c124c80..6cff358 100644
--- a/hals/hwc3/DisplayFinder.h
+++ b/hals/hwc3/DisplayFinder.h
@@ -22,10 +22,11 @@
 
 #include "Common.h"
 #include "DisplayConfig.h"
-#include "DrmClient.h"
 
 namespace aidl::android::hardware::graphics::composer3::impl {
 
+class DrmClient;
+
 struct DisplayMultiConfigs {
     int64_t displayId;
     int32_t activeConfigId;
diff --git a/hals/hwc3/FrameComposer.h b/hals/hwc3/FrameComposer.h
index 5e54561..72be5bc 100644
--- a/hals/hwc3/FrameComposer.h
+++ b/hals/hwc3/FrameComposer.h
@@ -26,11 +26,11 @@
 
 #include "Common.h"
 #include "DisplayChanges.h"
-#include "DrmClient.h"
 
 namespace aidl::android::hardware::graphics::composer3::impl {
 
 class Display;
+class DrmClient;
 
 class FrameComposer {
    public:
-- 
2.47.3
